DllImportGenerator/DllImportGenerator/DllImportStub.cs

```diff
@@ -203,7 +205,7 @@ public class GeneratedDllImportData
             }
 
             // Generate stub code
-            var stubGenerator = new StubCodeGenerator(method, dllImportData, paramsTypeInfo, retTypeInfo, diagnostics);
+            var stubGenerator = new StubCodeGenerator(method, dllImportData, paramsTypeInfo, retTypeInfo, diagnostics, env.Options);
```

Comment:
Do we want to expose the `StubEnvironment` on the `StubCodeContext`? This is another option rather than adding the analyzer options to the context.